### DISCUSSION TITLE
Should API projects generated with arcus-webapi template use Newtonsoft.Json instead of Json.Text

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.templates#277] [Q&A]
@fgheysels :

It seems that the new Json.Text serializer that is used by default in ASP.NET has a lot of limitations as opposed to the Newtonsoft.Json serializer.

I'm wondering if we shouldn't specify the Newtonsof.Json serializer for projects that are created with arcus-webapi.

Some limitations where I ran into today:

no out of the box support to serialize/deserialize Timespan types. You have to create your own converter for it. (Not such a big issue but ...)
No support for EnumMember attributes to control how enum values should be serialized. There is a workaround for it, but it's yet another external package.
Swashbuckle example's don't seem to work well when using the Json.Text serializer as well. When creating custom examples for a type which has enum properties, those enum-values are not serialized as a string although you have specified that enums should be formatted as a string. (Ok, this is most likely a problem in Swashbuckle.Filters, but it is confusing :) ).
So the question is, should we make sure that we use Newtonsoft.Json as the JSON serializer by specifying:

```csharp
 services.AddControllers(options =>
{
   ...
}).AddNewtonsoftJson(o =>
{
    o.SerializerSettings.Converters.Add(new StringEnumConverter());
});
```

Original issue: https://github.com/arcus-azure/arcus.templates/issues/159

**Comment by pim-simons** [comment]
Not sure if this is still relevant, but from your text above I only see upside to using Newtonsoft.Json as the JSON serializer, are there any downsides to using Newtonsoft.Json as the JSON serializer?

**Reply by tomkerkhove** [reply]
Yes, it's outdated and no longer maintained because Microsoft hired him to work on System.Text.Json.

However, in .NET 3.x it's not mature enough but didn't use .NET 5 yet.
